packages/ui/src/components/TradingForm.tsx

```diff
@@ -0,0 +1,106 @@
+import { Button, FormikTextField } from '@uzh-bf/design-system'
+import { Form, Formik } from 'formik'
+import * as yup from 'yup'
+
+// TODO(JJ):
+// - UserNotification as child?
+// - Name is currently Trading -> input spotPrice, how to name?
+
+function optionalValueToCHFString(value: number, digits = 2) {
+  return value?.toLocaleString('de-CH', {
+    style: 'currency',
+    currency: 'CHF',
+    maximumFractionDigits: digits,
+  })
+}
+
+interface Props {
+  price: number
+  nameButtonBuy: string
+  nameButtonSell: string
+  onSubmit: (values: any, helpers: any) => Promise<void>
+  min?: number
+  disableButtonBuy?: boolean
+  disableButtonSell?: boolean
+}
+
+function TradingForm({
+  price,
+  nameButtonBuy,
+  nameButtonSell,
+  onSubmit,
+  min = 0,
+  disableButtonBuy = false,
+  disableButtonSell = false,
+}: Props) {
+  return (
+    <div className="flex w-max gap-4 rounded border p-8">
+      <Formik
+        initialValues={{
+          modifier: 1,
+          volume: min,
+        }}
+        isInitialValid={false}
+        validationSchema={yup.object({
+          volume: yup.number().min(min).required(),
```

Comment:
add max also? e.g., for the case when you cannot sell more than you have in storage. we use that in derivatives game